app/src/main/java/com/kickstarter/ui/activities/LoginActivity.kt

```diff
@@ -3,129 +3,200 @@ package com.kickstarter.ui.activities
 import com.kickstarter.ui.data.ActivityResult.Companion.create
 import com.kickstarter.ui.data.LoginReason
 import com.kickstarter.ui.extensions.finishWithAnimation
-import com.kickstarter.ui.extensions.hideKeyboard
-import com.kickstarter.ui.extensions.onChange
-import com.kickstarter.ui.extensions.showSnackbar
-import com.kickstarter.ui.extensions.text
-import com.kickstarter.ui.views.ConfirmDialog
+import com.kickstarter.ui.extensions.startDisclaimerActivity
 import com.kickstarter.viewmodels.LoginViewModel
 import io.reactivex.android.schedulers.AndroidSchedulers
 import io.reactivex.disposables.CompositeDisposable
 
-class LoginActivity : AppCompatActivity() {
+class LoginActivity : ComponentActivity() {
 
-    private var confirmResetPasswordSuccessDialog: ConfirmDialog? = null
     private lateinit var ksString: KSString
 
-    private val forgotPasswordString = R.string.login_buttons_forgot_password_html
-    private val forgotPasswordSentEmailString = R.string.forgot_password_we_sent_an_email_to_email_address_with_instructions_to_reset_your_password
-    private val resetPasswordSentEmailString = R.string.forgot_password_we_sent_an_email_to_email_address_with_instructions_to_set_your_password
+    private val resetPasswordSentEmailString =
+        R.string.forgot_password_we_sent_an_email_to_email_address_with_instructions_to_set_your_password
     private val loginDoesNotMatchString = R.string.login_errors_does_not_match
     private val unableToLoginString = R.string.login_errors_unable_to_log_in
-    private val loginString = R.string.login_buttons_log_in
-    private val errorTitleString = R.string.login_errors_title
-    private lateinit var binding: LoginLayoutBinding
 
     private lateinit var viewModelFactory: LoginViewModel.Factory
     private val viewModel: LoginViewModel.LoginViewModel by viewModels { viewModelFactory }
 
     private lateinit var disposables: CompositeDisposable
 
+    private var darkModeEnabled = false
+    private var currentEmail = ""
+    private var currentPassword = ""
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         disposables = CompositeDisposable()
 
         val env = this.getEnvironment()?.let { env ->
             viewModelFactory = LoginViewModel.Factory(env, intent = intent)
+            darkModeEnabled =
+                env.featureFlagClient()?.getBoolean(FlagKey.ANDROID_DARK_MODE_ENABLED) ?: false
             env
         }
 
-        binding = LoginLayoutBinding.inflate(layoutInflater)
-
-        setContentView(binding.root)
+        setContent {
+            var prefillEmail = viewModel.outputs.prefillEmail().subscribeAsState(initial = "").value
+
+            var isLoading = viewModel.isLoading().subscribeAsState(initial = false).value
+
+            var error by remember { mutableStateOf("") }
+
+            errorMessages().subscribe {
+                error = it
+            }.addToDisposable(disposables)
```

Comment:
`            var error by remember { mutableStateOf("") }

            errorMessages().subscribe {
                error = it
            }.addToDisposable(disposables)`

Curious about what this block of code is doing?

Reply:
So this class was doing error messages in a different way, the error was not being put into the observable itself, but it was being mapped from a Unit observable, so I made a separate variable to hold the error so I can reset it, since I could not do so with the observable itself.

Let me know if that makes sense!